src/main.rs

```diff
@@ -1,40 +1,134 @@
 extern crate clap;
 extern crate ignore;
+extern crate regex;
+extern crate difference;
+extern crate term;
 
 use clap::{App, Arg};
 use ignore::Walk;
+use std::path::Path;
+use std::fs::File;
+use std::io::prelude::*;
+use regex::Regex;
+use difference::{Changeset, Difference};
+
+#[allow(unused_must_use)]
+fn term_show_diff(before: &str, after: &str) {
+   // Compare both texts, the third parameter defines the split level.
+    let Changeset { diffs, .. } = Changeset::new(before, after, "\n");
+
+    let mut t = term::stdout().unwrap();
+
+    for i in 0..diffs.len() {
+        match diffs[i] {
+            Difference::Same(ref _x) => {
+                t.reset().unwrap();
+                writeln!(t, "");
+            }
+            Difference::Add(ref x) => {
+                t.fg(term::color::GREEN).unwrap();
+                writeln!(t, "+{}", x);
+            }
+            Difference::Rem(ref x) => {
+                t.fg(term::color::RED).unwrap();
+                writeln!(t, "-{}", x);
+            }
+        }
+    }
+    t.reset().unwrap();
+    t.flush().unwrap();
+}
+
+
+pub struct Replacer {
+    re: Regex,
+    replacement: String,
+}
+
+impl Replacer {
+    pub fn new(pattern: &str, replacement: &str) -> Replacer {
+        let re = Regex::new(pattern).expect("aie aie aie");
```

Comment:
no french

Reply:
How should we handle error?

this panic... Can a constructor return a Result<Replacer> ?

Reply:
fixed